x-pack/packages/ai-infra/product-doc-artifact-builder/src/tasks/process_documents.ts

```diff
@@ -56,12 +56,21 @@ const processDocument = (document: ExtractedDocument) => {
     })
     // remove edit links
     .replaceAll(/\[\s*edit\s*\]\(\s*[^)]+\s*\)/g, '')
-    // remove empty links
-    .replaceAll('[]()', '')
+    // remove empty and image links
+    .replaceAll(/\[\]\(\s*[^)]+\s*\)/g, '')
```

Comment:
The old replacer matched `[]()`, but the new regex does not. You have to add _something_ within the parenthesis in order for that to work. Is that intentional?

```sh
> /\[\]\(\s*[^)]+\s*\)/g.test('[]()')
false
> /\[\]\(\s*[^)]+\s*\)/g.test('[]( )')
true
```